### PR TITLE
Use argument span for inlined singleton an type arguments

### LINKED ISSUE #6026: Wrong position for inlined pure arguments
```scala
class Foo {
  inline def fun(x: String): String = x
  def foo = fun("abc")
}
```

```
sbt:dotty> dotc Foo.scala -Xprint:front -Yprint-pos -Ydebug-pos
[warn] Multiple main classes detected.  Run 'show discoveredMainClasses' to see the list
[info] Running (fork) dotty.tools.dotc.Main -classpath /Users/nicolasstucki/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.12.8.jar:/Users/nicolasstucki/GitHub/dotty/library/../out/bootstrap/dotty-library-bootstrapped/scala-0.14/dotty-library_0.14-0.14.0-bin-SNAPSHOT.jar Foo.scala -Xprint:front -Yprint-pos -Ydebug-pos
result of Foo.scala after frontend:
package <empty>@Foo.scala<0..0> {
  class Foo() extends Object@Foo.scala<6..6>@
    Foo.scala<6..6>
  ()@Foo.scala<6..6> { 
    inline def fun(
      x: String@Foo.scala<32..38>@
        Foo.scala[29..38]
    ): String@Foo.scala<41..47> = 
      x@Foo.scala<50..51> :
        String@Foo.scala<41..47>
      @Foo.scala<41..51>
    @Foo.scala[14..25..51]
    def foo: String = 
      /* inlined from 
        this@Foo.scala<64..64>.fun@
          Foo.scala<64..67>
        ("abc"@Foo.scala<68..73>)@
          Foo.scala<64..74>
       */ 
        {
          "abc"@Foo.scala<50..51> :
            String@Foo.scala<41..47>
          @Foo.scala<64..74>
        }
      @Foo.scala<64..74>
    @Foo.scala[54..58..74]
  }@Foo.scala[0..6..76]
}@Foo.scala<0..76>
```

We can see that the position before inlining was `"abc"@Foo.scala<68..73>` but after inlining it becomes `"abc"@Foo.scala<50..51>` which is the position of the `x` in the body of `fun`.

#6024 will add a way to test these positions.

### PR DESCRIPTION
**Pull request by nicolasstucki**
Fixes #6026 and #6027 by using the span of the inlined argument for the tree of a parameter reference.

### CODE REVIEW

File: compiler/test/dotty/tools/backend/jvm/InlineBytecodeTests.scala [hunk 1]
```diff
@@ -101,6 +101,399 @@ class InlineBytecodeTests extends DottyBytecodeTest {
...
+    checkBCode(source) { dir =>
+      val clsIn      = dir.lookupName("Foo.class", directory = false).input
+      val clsNode    = loadClassNode(clsIn, skipDebugInfo = false)
+
+      val track = clsNode.methods.asScala.find(_.name == "fact")
+      assert(track.isEmpty, "method `fact` should have been erased")
+
+      val main = getMethod(clsNode, "main")
+      val instructions = instructionsFromMethod(main)
+      val expected =
```
Comment (allanrenucci) [inline compiler/test/dotty/tools/backend/jvm/InlineBytecodeTests.scala]: Can we do better that these huge and unmaintainable bytecode tests? See e04918e3fe688883443f35659a74e85f5347856d. I suggest having tests that only assert the presence of the instructions you are expecting (I believe `LineNumber` here)

Comment (nicolasstucki) [reply compiler/test/dotty/tools/backend/jvm/InlineBytecodeTests.scala]: I will not keep this test. I added them to try to figure out the source of the bug I'm chasing.
Now I have a minimized version of it that will eventually replace it.

Comment (nicolasstucki) [reply compiler/test/dotty/tools/backend/jvm/InlineBytecodeTests.scala]: I opened #6085 with the minimized version of that issue

File: compiler/src/dotty/tools/dotc/typer/Inliner.scala [hunk 2]
```diff
@@ -439,12 +444,18 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(implicit ctx: Context) {
             case _ => tree
           }
         case tree: Ident =>
+          def span =
+            if (tree.name == nme.WILDCARD) tree.span // From type match
+            else if (tree.symbol.isTypeParam && tree.symbol.owner.isClass) tree.span // TODO is this the correct span?
+            else paramSpan(tree.name)
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/typer/Inliner.scala]: Maybe add one line document to `def span`?

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/typer/Inliner.scala]: Added a comment and renamed `span` to `argSpan` to make it clearer.

**Pull request merged**